src/networks/competetive_net.py

```diff
@@ -238,4 +238,107 @@ def compute_loss_against(self, opponent, input):
         outputs = sm(outputs_full[:, -1, :].contiguous().view(-1))
         d_loss_fake = self.loss_function(outputs, fake_labels)
 
-        return d_loss_real + d_loss_fake, None
+        return d_loss_real + d_loss_fake, None, None
+
+
+class SSDiscriminatorNet(DiscriminatorNet):
+
+    def __init__(self, adv_loss, label_pred_loss, num_classes, net, data_size,
+                 adv_layer, label_pred_layer, optimize_bias=True):
+        DiscriminatorNet.__init__(self, adv_loss, net, data_size, optimize_bias=optimize_bias)
+        self.num_classes = num_classes
+        self.adv_layer = adv_layer
+        self.label_pred_layer = label_pred_layer
+        self.label_pred_loss = label_pred_loss
+
+    @property
+    def name(self):
+        return 'SemiSupervisedDiscriminator'
+
+    @property
+    def default_fitness(self):
+        return float('-inf')
+
+    def clone(self):
+        return SSDiscriminatorNet(self.loss_function,
+                                  self.label_pred_loss,
+                                  self.num_classes,
+                                  copy.deepcopy(self.net),
+                                  self.data_size,
+                                  self.adv_layer,
+                                  self.label_pred_layer,
+                                  self.optimize_bias)
+
+    def compute_loss_against(self, opponent, input, labels=None):
+        """
+        Assumes input is provided as a list and that the last column is an
+        integer column representing the class of the corresponding input
+        """
+        # Compute CrossEntropyLoss using real images
+        batch_size = input.size(0)
+        tensor = torch.Tensor(batch_size)
+        tensor.fill_(self.num_classes)
+        tensor = tensor.long()
+        fake_labels = to_pytorch_variable(tensor)
+        real = to_pytorch_variable(torch.ones(batch_size))
```

Comment:
could you call them real_unsupervised_labels and fake_unsupervised_labels?? To make it more clear